src/components/accession2/view/Accession2View.tsx

```diff
@@ -302,20 +302,25 @@ export default function Accession2View(props: Accession2ViewProps): JSX.Element
             <ViewViabilityTestModal
               open={viewViabilityTestModalOpened}
               accession={accession}
-              onClose={() => setViewViabilityTestModalOpened(false)}
-              selectedTest={selectedTest}
-              setNewViabilityTestOpened={setNewViabilityTestOpened}
-              setSelectedTest={setSelectedTest}
+              viabilityTest={selectedTest}
+              onClose={() => {
+                setViewViabilityTestModalOpened(false);
+                setSelectedTest(undefined);
+              }}
+              onEdit={() => setNewViabilityTestOpened(true)}
             />
           )}
           <NewViabilityTestModal
             open={newViabilityTestOpened}
             reload={reloadData}
             accession={accession}
-            onClose={() => setNewViabilityTestOpened(false)}
+            onClose={() => {
+              setNewViabilityTestOpened(false);
+              setSelectedTest(undefined);
```

Comment:
I think this is not working correctly. If I click on add test, fill some fields and close the modal with Cancel, and then click "add test" again, the data I filled before is still in the modal

Reply:
But it's unrelated with this change, it wasn't cleaning the fields before either

Reply:
Yes, that is a good idea. I didn't realize we could do that to clean up

Reply:
Thanks, I will make that change.

Reply:
Made the change, if you get a chance to test let me know. It works great now. Noticed we had this problem in other modals as well like edit-state etc. All good now.

Reply:
> I was testing and on edit quantity modal, the link "go to weight calculator" is not working

Looks like we are calling onClose here, maybe we should introduce an on-hide here.

Reply:
actually, i can reuse the isCalculatorOpened flag

Reply:
I will just remove this check for quantity modal since it is already doing a cleanup on close.